app/assets/javascripts/admin/services/query_persistence.js.coffee

```diff
@@ -0,0 +1,30 @@
+angular.module("admin.indexUtils").factory 'QueryPersistence', (localStorageService)->
+  new class QueryPersistence
+    storageKey: ''
+    storableFilters: []
+
+    constructor: ->
+      localStorageService.setStorageType("sessionStorage")
+
+    getStoredFilters: ->
+      localStorageService.get(@storageKey) || {}
+
+    setStoredFilters: (scope) ->
+      filters = {}
+      for key in @storableFilters
+        filters[key] = scope[key]
+      localStorageService.set(@storageKey, filters)
+
+    restoreFilters: (scope) ->
+      storedFilters = @getStoredFilters()
+
+      unless _.isEmpty(storedFilters)
+        for k,v of storedFilters
+          scope[k] = v
+
+        return true
+
+      false
+
+    clearFilters: () ->
+      localStorageService.remove(@storageKey)
```

Comment:
I dont think this class should be specific to filters. I'd rename most of the variables and methods to make it a generic LocalStorage KeyValueMapStore, the code is exacly the same (I just used a guard clause on restoreValues method to make it simpler):

```js
angular.module("admin.indexUtils").factory 'KeyValueMapStore', (localStorageService)->
  new class KeyValueMapStore
    localStorageKey: ''
    storableKeys: []

    constructor: ->
      localStorageService.setStorageType("sessionStorage")

    getStoredKeyValueMap: ->
      localStorageService.get(@localStorageKey) || {}

    setStoredValues: (source) ->
      keyValueMap = {}
      for key in @storableKeys
        keyValueMap[key] = source[key]
      localStorageService.set(@localStorageKey, keyValueMap)

    restoreValues: (target) ->
      storedKeyValueMap = @getStoredKeyValueMap()

      return false unless _.isEmpty(storedKeyValueMap)  
      
      for k,v of storedKeyValueMap
       target[k] = v

      return true

    clearKeyValueMap: () ->
      localStorageService.remove(@localStorageKey)


```

Reply:
One way or another this is a nice and easy class to add some js unit tests.

Reply:
I think it's a very relevant idea, and commit 7008b5ec7 should take into account your comments ;) Thanks! 🙏

Reply:
awesome :ok_hand: